---
title: " Final"
author: "Vera Pinto & Carina Silva"
date: "2023-01-10"
output: html_document
---

```{r setup, echo=FALSE}
library(readxl)
NGS_Farmacogenetica_completa_original <- read_excel("clean.xlsx", sheet="clean")
data2<-as.data.frame(NGS_Farmacogenetica_completa_original)
names(data2)
class(data2)
attach(data2)
```

```{r,echo=FALSE}
library(dplyr)
amostra.unicas<-data2 %>% distinct(data2$Sample, .keep=data2$Variant)
str(amostra.unicas)
dim(amostra.unicas)
amostra.unicas<-as.data.frame(amostra.unicas)
                      

coordenadas.unicas<-data2 %>% distinct(data2$Coordinate, .keep=data2$Chr)
names(coordenadas.unicas)
dim(coordenadas.unicas)
class(coordenadas.unicas)

classification.unicas<-data2 %>% distinct(data2$Classification, .keep=data2$Sample)
names(classification.unicas)
table(classification.unicas$`data2$Classification`)
class(classification.unicas)
dim(classification.unicas)

snp.unicas<-data2 %>% distinct(data2$SNP, .keep=data2$Coordinate)
names(snp.unicas)
table(snp.unicas$`data2$SNP`)
class(snp.unicas)
dim(snp.unicas)

gen.unicas<-data2 %>% distinct(data2$Genotype, .keep=data2$Sample)
names(gen.unicas)
table(gen.unicas$`data2$Genotype`)
class(gen.unicas)
dim(gen.unicas)

rec.unicas<-data2 %>% distinct(data2$Recess, .keep=data2$Sample)
names(rec.unicas)
table(rec.unicas$`data2$Recess`)
class(rec.unicas)
dim(rec.unicas)

snpid.unicas<-data2 %>% distinct(data2$SNPID, .keep=data2$Coordinate)
names(snpid.unicas)
table(snpid.unicas$`data2$SNPID`)
class(snpid.unicas)
dim(snpid.unicas)

# Alg 1 

#creating an alghorithm to find the coordinates who didi not have variants identified, working as our 0's
#the data obtained form virtual studio of ilumina only annotates the coordinates whent there is a variant identified
#ommitng the coordinates when there aren't, so its crucial to identified them

coordenada_lista_amostras2<-function(coordenadas.unicas,amostra.unicas,data2) {
  
  linhas_novas<-data.frame(matrix(ncol =9, nrow = 0))
  data_final<-data.frame(matrix(ncol = 9, nrow = 0))
  

  
  for (i in 1:dim(coordenadas.unicas)[1]){
   
 # identifies the samples associated with a coordinate
    lista_samples<-as.data.frame(data2$Sample[data2$Coordinate==coordenadas.unicas[i,1]])
    
    #set of samples that are not associated with a coordinate
    out<-setdiff(amostra.unicas[,1],lista_samples[,1])
    
        if (length(out)!=0){
          
          classif<-data2[data2$Sample %in% out,]
          classif2<-classif %>% distinct(classif$Sample, .keep=classif$Classification)
          classif2<-as.data.frame(classif2)
          snp<-data2[data2$Sample %in% out,]
          snp2<-snp %>% distinct(snp$Sample, .keep=snp$SNP)
          snp2<-as.data.frame(snp2)
          gen<-data2[data2$Sample %in% out,]
          gen2<-gen %>% distinct(gen$Sample, .keep=gen$Genotype)
          gen2<-as.data.frame(gen2)
          rec<-data2[data2$Sample %in% out,]
          rec2<-rec %>% distinct(rec$Sample, .keep=gen$Recess)
          rec2<-as.data.frame(rec2)
          snpid<-data2[data2$Sample %in% out,]
          snpid2<-snp %>% distinct(snpid$Sample, .keep=snp$SNPID)
          snpid2<-as.data.frame(snpid2)
          linhas_novas<-data.frame(Sample=out,Chr=rep(coordenadas.unicas[i,2],length(out)),Coordinate=rep(coordenadas.unicas[i,1],length(out)),
                                     Classification=classif2[,2],Variant=rep(0,length(out)), SNP=rep(snp.unicas[i,1],length(out)), Genotype="hom",Recess=0,SNPID=rep(snpid.unicas[i,1],length(out)))
    
          data_final<-rbind(data_final,linhas_novas)
          
                            } 
    
    else i<-i+1
    
  } 
  data_final <- data_final %>% distinct()
  return(data_final)
}


LISTA2<-coordenada_lista_amostras2(coordenadas.unicas,amostra.unicas,data2)


#juntar LISTA com Dados

FINAL2<-rbind(data2,LISTA2)
dim(FINAL2)
```

