---
title: "Algor√≠tmo 2"
author: "Vera Pinto & Carina Silva"
date: "2022-12-06"
output: html_document
---

```{r setup, include=FALSE}
library(readxl)
library(dplyr)
farmacog_NGS_FINAL <- read_excel("C:/Users/pinto/OneDrive - Universidade de Lisboa/Artigo/farmacog1.xlsx")
data<- farmacog_NGS_FINAL
```

```{r,include=FALSE}
results<-data.frame(matrix(ncol = 4, nrow = 0)) #data frame final
  cols <- (c("Chr", "SNP","Coordinate","p-value"))
  colnames(results) <- cols
  

  for (i in 1:23){
    if (i==21){next}
    sub <- subset(data, Chr==i)
    coor.unic<-sub %>% distinct(sub$Coordinate)
              for (j in  1:nrow(coor.unic)) {
                      coor<-coor.unic$`sub$Coordinate`[j]
                      SNP<-sub$SNP[j]
                      sub2 <- subset(sub, Coordinate==coor)
                      t<-table(sub2$Variant,sub2$Classification)
      
                            if (dim(t)[1]==2 & dim(t)[2]==2){
         
                              x<-fisher.test(t)
                              v<-x$p.value
                              v1<-data.frame(t(matrix(c(sub$Chr[i],SNP,coor,v))))
                              colnames(v1) <- cols
                              results<-rbind(results,v1)
                                         }
      
        
            }
     
    
    }
  
```





```{r}
write.csv(results, "C:/Users/pinto/OneDrive - Universidade de Lisboa/Artigo/results3.csv", row.names=FALSE)
```


```{r}
results
```

```{r}
library(qqman)
```

```{r}
results3 <- read_excel("C:/Users/pinto/OneDrive - Universidade de Lisboa/Artigo/results3.xlsx")
```

```{r}
manhattan(results2,chr = "Chr", bp="BP", p="p-value" , snp = "SNP",  
 annotatePval = 0.05,suggestiveline=TRUE, genomewideline=TRUE,   col = c("cyan", "blue","blueviolet"))
```
```{r}
p.adjust(results2$`p-value`, method = "holm", n=length(results2$`p-value`))
```
```{r}
manhattan(results3,chr = "Chr", bp="Coordinate", p="p-value" , snp = "SNP",  
 annotatePval = 0.05,suggestiveline=TRUE, genomewideline=TRUE,   col = c("cyan", "blue","blueviolet"))
```


```{r}
results2<-data.frame(matrix(ncol = 4, nrow = 0)) #data frame final
  cols <- (c("Chr", "SNP","Coordinate","p-value"))
  colnames(results2) <- cols
  

  for (i in 1:23){
    if (i==21){next}
    sub <- subset(data, Chr==i)
    coor.unic<-sub %>% distinct(sub$Coordinate)
              for (j in  1:nrow(coor.unic)) {
                      coor<-coor.unic$`sub$Coordinate`[j]
                      SNP<-sub$SNP[j]
                      sub2 <- subset(sub, Coordinate==coor)
                      t<-table(sub2$Recess,sub2$Classification)
      
                          if (dim(t)[1]==2 & dim(t)[2]==2){
         
                            x<-fisher.test(t)
                            v<-x$p.value
                            v1<-data.frame(t(matrix(c(sub$Chr[i],SNP,coor,v))))
                            colnames(v1) <- cols
                            results2<-rbind(results2,v1)
                                         }
      
        
            }
     
    
    }
```

```{r}
results2

```

```{r}
write.csv(results2, "C:/Users/pinto/OneDrive - Universidade de Lisboa/Artigo/results_recess.csv", row.names=FALSE)
```

```{r}
results2 <- read_excel("C:/Users/pinto/OneDrive - Universidade de Lisboa/Artigo/results_recess.xlsx")

manhattan(results2,chr = "Chr", bp="Coordinate", p="p-value" , snp = "SNP",  
 annotatePval = 0.05,suggestiveline=TRUE, genomewideline=TRUE,   col = c("cyan", "blue","blueviolet"))

```
```{r}
p.adjust(results2$`p-value`, method = "holm", n=length(results2$`p-value`))


```

